tests/test_zenhub.py

```diff
@@ -123,12 +127,106 @@ def test_nonexisting_repo_ID_nonexisting_issue_num(self, mocked_get_info, mock_r
         self.assertEqual(res.get_info(), {'message': 'Invalid Field for repo_id: repo_id is a required field'},
                          'get_info has incorrect output')
 
+    @patch('src.zenhub.ZenHub._get_repo_id')
     @patch('src.zenhub.ZenHub._generate_url', return_value='https://foo.bar')
-    def test_generate_url(self, mock_generate_url):
+    def test_generate_url(self, mock_generate_url, mock_repo_id):
 
+        mock_repo_id.return_value = '100000000'  # Needed for generating repo_ids in init.
         zen = ZenHub(repo_name='baz', issue=42)
         self.assertEqual(zen.url, 'https://foo.bar', 'URL not generated correctly')
 
+    @patch('os.path.join')
+    @patch('requests.put')
+    def test_update_issue_points(self, mock_put_change_points, mock_url_creator):
+        """Test that ZenHub.update_issue_points() works."""
+        issue_num = 42
+        new_points = 3
+        mock_url_creator.return_value = f'https://api.zenhub.io/p1/repositories/issues/{issue_num}/estimate'
+
+        zen = ZenHub(repo_name='azul', issue=issue_num)
+        zen._update_issue_points(new_points)
+
+        mock_put_change_points.assert_called()
+        request_args = list(mock_put_change_points.call_args)
+
+        # Check that the url is in the request
+        self.assertIn((mock_url_creator.return_value,), request_args)  # MagicMock stores this as a tuple.
+
+        # Check that the json_dict is in the put request.
+        expected_dict = {'headers': zen.headers.copy()}
+        expected_dict.update({'json': {'estimate': new_points}})
+        self.assertIn(expected_dict, request_args)
+
+    @patch('os.path.join')
+    @patch('requests.post')
+    def test_update_issue_pipeline(self, mock_post_change_pipeline, mock_url_creator):
+        """Test that ZenHub.update_issue_pipeline() works."""
+        issue_num = 42
+        mock_url_creator.return_value = f'https://api.zenhub.io/p1/repositories/issues/{issue_num}/moves'
+
+        zen = ZenHub(repo_name='azul', issue=issue_num)
+        zen.pipeline_ids = {'Icebox': 12345}
+        zen._update_issue_pipeline('Icebox')
+
+        mock_post_change_pipeline.assert_called()
+        request_args = list(mock_post_change_pipeline.call_args)
+
+        # Check that the url is in the request
+        self.assertIn((mock_url_creator.return_value,), request_args)  # MagicMock stores this as a tuple.
+
+        # Check that the json_dict is in the put request.
+        expected_dict = {'headers': zen.headers.copy()}
+        expected_dict.update({'json': {'pipeline_id': 12345, 'position': 'top'}})
+        self.assertIn(expected_dict, request_args)
+
```

Comment:
I don't see a test for `pipeline_pos`. Is this a priority rank of an issue within a pipeline?